pwiz_tools/Skyline/EditUI/PasteDlg.cs

```diff
@@ -444,57 +459,48 @@ private SrmDocument AddPeptides(SrmDocument document, bool validating, ref Ident
-                if (string.IsNullOrEmpty(peptideSequence) && string.IsNullOrEmpty(proteinName))
-                {
-                    continue;
-                }
-                if (string.IsNullOrEmpty(peptideSequence))
+                var sequenceAdductTuple = FastaSequence.ParsePeptideSequenceAndAdduct(
+                    Convert.ToString(row.Cells[colPeptideSequence.Index].Value), out string errorMessage);
+                if (errorMessage != null)
                 {
                     ShowPeptideError(new PasteError
                     {
                         Column = colPeptideSequence.Index,
                         Line = i,
-                        Message = Resources.PasteDlg_ListPeptideSequences_The_peptide_sequence_cannot_be_blank
+                        Message = errorMessage
                     });
                     return null;
                 }
 
-                CrosslinkLibraryKey crosslinkLibraryKey =
-                    CrosslinkSequenceParser.TryParseCrosslinkLibraryKey(peptideSequence, 0);
-                if (crosslinkLibraryKey == null)
+                var peptideSequence = sequenceAdductTuple.Item1;
+                var adduct = sequenceAdductTuple.Item2;
+                var proteinName = Convert.ToString(row.Cells[colPeptideProtein.Index].Value);
+                if (string.IsNullOrEmpty(peptideSequence) && string.IsNullOrEmpty(proteinName))
                 {
-                    if (!FastaSequence.IsExSequence(peptideSequence))
-                    {
-                        ShowPeptideError(new PasteError
-                        {
-                            Column = colPeptideSequence.Index,
-                            Line = i,
-                            Message = Resources.PasteDlg_ListPeptideSequences_This_peptide_sequence_contains_invalid_characters
-                        });
-                        return null;
-                    }
-                    peptideSequence = FastaSequence.NormalizeNTerminalMod(peptideSequence);
+                    listSequences.Add(null);
+                    continue;
                 }
-                else if (!crosslinkLibraryKey.IsSupportedBySkyline())
+                if (string.IsNullOrEmpty(peptideSequence))
                 {
                     ShowPeptideError(new PasteError
                     {
                         Column = colPeptideSequence.Index,
                         Line = i,
-                        Message = Resources.PasteDlg_ListPeptideSequences_The_structure_of_this_crosslinked_peptide_is_not_supported_by_Skyline
+                        Message = Resources.PasteDlg_ListPeptideSequences_The_peptide_sequence_cannot_be_blank
                     });
                     return null;
                 }
-
-                listSequences.Add(peptideSequence);
+                listSequences.Add(Tuple.Create(peptideSequence, adduct));

```

Comment:
Why not just use sequenceAdductTuple? I don't see that the values get changed anywhere. If you used a struct instead of a Tuple, the code would likely read clearly enough without needing to create the named variables.